providers/google.go

```diff
@@ -229,37 +229,73 @@ func (p *GoogleProvider) setGroupRestriction(opts options.GoogleOptions) {
 	}
 }
 
-func getAdminService(opts options.GoogleOptions) *admin.Service {
-	ctx := context.Background()
-	var client *http.Client
+// https://developers.google.com/admin-sdk/directory/reference/rest/v1/members/hasMember#authorization-scopes
+var possibleScopesList = [...]string{
+	admin.AdminDirectoryGroupMemberReadonlyScope,
+	admin.AdminDirectoryGroupReadonlyScope,
+	admin.AdminDirectoryGroupMemberScope,
+	admin.AdminDirectoryGroupScope,
+}
+
+func getOauth2TokenSource(ctx context.Context, opts options.GoogleOptions, scope string) oauth2.TokenSource {
 	if opts.UseApplicationDefaultCredentials {
 		ts, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
 			TargetPrincipal: getTargetPrincipal(ctx, opts),
-			Scopes:          []string{admin.AdminDirectoryGroupReadonlyScope, admin.AdminDirectoryUserReadonlyScope},
+			Scopes:          []string{scope},
```

Comment:
Is this really desired to reduce the scopes list to a single scope without the option for parsing in multiple values anymore?

What about existing clients that have configured their oauth2 applications with the scopes:

```
    https://www.googleapis.com/auth/admin.directory.group.readonly
    https://www.googleapis.com/auth/admin.directory.user.readonly
```

This new implementation would create a client with the scope `admin.directory.group.readonly` only (for existing applications) wouldn't this be cause for a breaking change?

Reply:
Existing clients will continue to work normally.

We define a `possibleScopesList`, which means that as long as one of the scopes is available, it can work normally.

For existing clients, `https://www.googleapis.com/auth/admin.directory.group.readonly` must have been configured. It is also in `possibleScopesList`, so it will not be a breaking change.

The purpose of this PR is to support new clients to configure only a smaller scope of permissions `https://www.googleapis.com/auth/admin.directory.group.member.readonly
` and tell everyone how to configure it in the document.

Reply:
I got it but doesn't the request to google only request a client with only admin.directory.group.readonly? I would expect that the resulting client will only have the permissions to interact with groups but not users?

Reply:
> I would expect that the resulting client will have the permissions to interact with groups but not with user data?

Yes, this permission is sufficient for our use case.

In our scenario, the user name and group name will be given in advance. Based on these two determined values, we need to determine whether the user is in the group.